### PR TITLE
Fixed full options mode

### PR DESCRIPTION
**Pull request by lparsons**
Full option selectors where checking for "True" but truevalues were
"set" and thus no options were actually being passed to the command
line. Also updated to conform to best practices by converting the
dynamic options booleans to selects.

### CODE REVIEW

**Comment by lparsons** [comment]
@guerler @bgruening mentioned this issue might be due to a tool-form bug. Just thought you might want to look into it. @nekrut Have you see the issue when using "full options" before?

**Comment by nsoranzo** [comment]
@lparsons The last test fails on Travis.

**Comment by lparsons** [comment]
@nsoranzo The tests pass for me locally. Is there a way to get the output from the travis test to see what's wrong?

**Comment by lparsons** [comment]
@nsoranzo OK, looks like a rounding issue between platforms, I removed that specific line in the output and all it well.

**Comment by nsoranzo** [comment]
:+1:  Thanks @lparsons!

I'll leave it to @bgruening for final review and merge.

File: tools/freebayes/freebayes.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<tool id="freebayes" name="FreeBayes" version="1.0.2.29">
+<tool id="freebayes" name="FreeBayes" version="1.0.2.29_galaxy1">
```
Comment (bgruening) [inline tools/freebayes/freebayes.xml]: General question what do people think about `_1`, `_2` or `--1` and `--2` similar to conda build numbers?
I don't like this galaxy on, especially not if we want to have here a Galaxy independent tool description.

Comment (bgruening) [inline tools/freebayes/freebayes.xml]: ping @galaxyproject/iuc

File: tools/freebayes/freebayes.xml [hunk 2]
```diff
@@ -617,9 +650,17 @@
             <param name="input_bams" ftype="bam" value="freebayes-phix174.bam"/>
             <param name="options_type_selector" value="naive_w_filters"/>
             <param name="min_coverage" value="14"/>
-            <output name="output_vcf" file="freebayes-phix174-test3.vcf" lines_diff="2"/>
+            <output name="output_vcf" file="freebayes-phix174-test3.vcf" compare="contains"/>
+        </test>
+        <test>
+            <param name="reference_source_selector" value="history" />
```
Comment (bgruening) [inline tools/freebayes/freebayes.xml]: Yeah, new test! Fantastic!

File: tools/freebayes/freebayes.xml [hunk 3]
```diff
@@ -525,8 +552,11 @@
                 </conditional>
                 <!-- algorithmic features -->
                 <conditional name="algorithmic_features">
-                    <param name="algorithmic_features_selector" type="boolean" truevalue="set" falsevalue="do_not_set" label="Tweak algorithmic features?"
-                        help="Sets --report-genotypes-likelihood-max, -B, --genotyping-max-banddepth, -W, -N, S, -j, -H, -D, -= options" />
+                    <param name="algorithmic_features_selector" type="select" label="Algorithmic features"
+                        help="Sets --report-genotypes-likelihood-max, -B, --genotyping-max-banddepth, -W, -N, S, -j, -H, -D, -= options">
```
Comment (bgruening) [inline tools/freebayes/freebayes.xml]: @lparsons do you know what `-=` is?

File: tools/freebayes/freebayes.xml (hunk 1 above)
Comment (lparsons) [inline tools/freebayes/freebayes.xml]: I'm fine with whatever the group decides, though I don't understand "Galaxy independent tool description". This version specifically refers to the Galaxy wrapper.

File: tools/freebayes/freebayes.xml (hunk 3 above)
Comment (lparsons) [inline tools/freebayes/freebayes.xml]:
```
   -= --genotype-qualities
                   Calculate the marginal probability of genotypes and report as GQ in
                   each sample field in the VCF output.
```

Comment (bgruening) [inline tools/freebayes/freebayes.xml]: This looks really unusual. Thanks for checking.

**Comment by bgruening** [comment]
Thanks @lparsons!!!
Btw. look at your private gitter massages :)

**Pull request merged**